_pytest/assertion/truncate.py

```diff
@@ -0,0 +1,102 @@
+"""
+Utilities for truncating assertion output.
+
+Current default behaviour is to truncate assertion explanations at
+~8 terminal lines, unless running in "-vv" mode or running on CI.
+"""
+
+import os
+
+import py
+
+
+DEFAULT_MAX_LINES = 8
+DEFAULT_MAX_CHARS = 8 * 80
+USAGE_MSG = "use '-vv' to show"
+
+
+def truncate_if_required(explanation, item, max_length=None):
+    """
+    Truncate this assertion explanation if the given test item is eligible.
+    """
+    if _should_truncate_item(item):
+        return _truncate_explanation(explanation)
+    return explanation
+
+
+def _should_truncate_item(item):
+    """
+    Whether or not this test item is eligible for truncation.
+    """
+    verbose = item.config.option.verbose
+    return verbose < 2 and not _running_on_ci()
+
+
+def _running_on_ci():
```

Comment:
@The-Compiler cool, thanks.

> it's just a sane default IMHO (because you can't easily debug failures on CI

This is a good point, but I also agree with @flub that it's quite hidden / implicit behaviour. Do you think it's worth adding `--help` documentation for this? Or moving it up to config.py and linking it to -vv?

Either way I think this is a separate feature that should be changed (or not) on another branch.